mobile_verifier/src/heartbeats/mod.rs

```diff
@@ -293,8 +296,37 @@ impl HeartbeatReward {
         }
     }
 
-    pub fn reward_weight(&self) -> Decimal {
-        self.location_trust_score_multiplier
+    pub fn trust_score_multiplier(&self, overlaps_boosted: bool) -> Decimal {
+        if self.cbsd_id.is_some() {
+            // If this is a cbrs radio, the trust score is always 1
+            return dec!(1.0);
+        }
+        if overlaps_boosted {
+            // If we overlap a boosted hex, use the more restrictive distance
+            // check:
+            let distances = self.distances_to_asserted.as_ref().unwrap();
```

Comment:
It will never be none. One of the migrations adds a never null check to the distance to asserted column, and `array_agg` only returns NULL when all of the columns are null

Reply:
Caveat: it will be none if the radio is CBRS